### PR TITLE
Add first post guided tour

### PR DESCRIPTION
**Pull request by markryall**
This PR adds a guided tour triggered from the onboarding checklist.

# Testing

* Visit `/checklist/<domain>` for your site
* Click 'Do It' for the first post task (whether completed or not - the tour should show)
* Verify that the following steps appear in the tour:

first-post

### CODE REVIEW

File: client/layout/guided-tours/tours/checklist-publish-post-tour.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+		<Step
+			name="featured-images"
+			target="editor-featured-image-current-image"
+			arrow="top-left"
+			placement="below"
+		>
+			<Continue target="editor-featured-image-current-image" step="choose-image" click>
+				<p>
+					{ translate(
+						'Featured images are a great way to add more personality to your pages. ' +
```
Comment (a8ci18n) [inline client/layout/guided-tours/tours/checklist-publish-post-tour.js]: Hi! I've found a possible matching string that has already been translated **1** time:
`translate( 'Featured images are a great way to add more personality to your pages. Let's add something a little more relevant to your About page text.' )` *ES Score: 15*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/first-post-tour)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by lsinger** [comment]
I don't get a "do it" link as I have a post already:

screen shot 2017-12-19 at 12 01 55 pm

Instead it gives me an "edit" link which opens the editor but to an invalid post (and it also starts the tour, but that's not helpful at that time because of the editor's error modal): 

screen shot 2017-12-19 at 12 02 50 pm

**Comment by lsinger** [comment]
The tags step doesn't check for the sidebar being open:

screen shot 2017-12-19 at 12 04 15 pm

**Comment by lsinger** [comment]
The featured images step doesn't work ... there's no "featured image" area there. 

screen shot 2017-12-19 at 12 04 30 pm

**Comment by fditrapani** [comment]
> The tags step doesn't check for the sidebar being open:

@lsinger can we add logic to check if it's open or open it if it's closed?

**Comment by lsinger** [comment]
> @lsinger can we add logic to check if it's open or open it if it's closed?

(addressed in Slack I think)

**Comment by fditrapani** [comment]
Some good points here raised by Leif. I know this is a bit of a hack but maybe for the short term, we don't display any tours when you go back and edit a completed task? 

In the perfect conditions, new site with not edits done to the content, this works well. 

A nice to have would would be if we could open the tags container when the tour points there. @lsinger is that possible? This could be done in another PR though. No need to add that complexity here.

**Comment by taggon** [comment]
Hey! As Mark and I discussed, I pushed my hack to this branch.

**Comment by fditrapani** [comment]
What ever you did @taggon, it worked really well on the featured image. We should add it to the Categories tour as well (check if the sidebar is open). Alternatively, does anyone know if we can open the sidebar in any way and still show the tour?

**Comment by fditrapani** [comment]
The Tour is looking good. I've created some Trello cards to keep track of the issues @lsinger shared. We can tackle them in separate PRs.

**Pull request merged**